### PR TITLE
Fix proc_regs to handle Cheri PC offset in normal registers

### PR DESCRIPTION
**Pull request by nicomazz**
In the normal registers gathered from `PT_GETREGS` (ptrace) the pc is only the offset.
To have the complete address we have to add the base of the PCC.

The rest of `libproc` assumes that the PC is the complete one, and not only the offset.

### CODE REVIEW

File: lib/libproc/proc_regs.c [hunk 1]
```diff
@@ -35,14 +35,32 @@ __FBSDID("$FreeBSD$");
...
+#include <cheri/cheric.h>
+#include <cheri/cheri.h>
+
+static int
+get_pcc_base(struct proc_handle *phdl)
+{
+	struct capreg capregs;
+
+	if (ptrace(PT_GETCAPREGS, proc_getpid(phdl), (caddr_t)&capregs, 0) < 0)
+		return (-1);
```
Comment (qwattash) [inline lib/libproc/proc_regs.c:58]: Returning `-1` without checking it at regvalue assignment may result in unexpected decrement/increment of `regvalue`.
Also `int` may be not wide enough for a vaddr, better use `vaddr_t`, perhaps you want to pass the `regvalue` argument and fill it directly with `(__cheri_addr vadd_t)pcc_value`?

**Comment by jrtc27** [comment]
There is the question of whether libproc seeing the PC as the architectural PC or the address is the right choice. In either case, why are we even querying the integer registers when everything we need is in PCC? If we do want the address, we should just get the address of PCC, not get PC and add PCC's base. You will also notice that PC is not the only register that's broken; on purecap MIPS, REG_SP is going to give you complete garbage, because $c11 is the stack pointer and $sp is completely unused.

**Comment by nicomazz** [comment]
Now the integer registers are not queried in the getregs case.

**Comment by arichardson** [comment]
Jenkins does not seem happy:
```
/local/scratch/jenkins/workspace/CheriBSD-pipeline_PR-472/cheribsd/lib/libproc/proc_regs.c:101:2: error: #elif without #if

#elif defined(__powerpc__)

 ^

/local/scratch/jenkins/workspace/CheriBSD-pipeline_PR-472/cheribsd/lib/libproc/proc_regs.c:102:20: error: no member named 'pc' in 'struct reg'

                *regvalue = regs.pc;

                            ~~~~ ^

/local/scratch/jenkins/workspace/CheriBSD-pipeline_PR-472/cheribsd/lib/libproc/proc_regs.c:103:2: error: #elif without #if

#elif defined(__riscv)

 ^

/local/scratch/jenkins/workspace/CheriBSD-pipeline_PR-472/cheribsd/lib/libproc/proc_regs.c:104:20: error: no member named 'sepc' in 'struct reg'

                *regvalue = regs.sepc;

                            ~~~~ ^

/local/scratch/jenkins/workspace/CheriBSD-pipeline_PR-472/cheribsd/lib/libproc/proc_regs.c:105:2: error: #endif without #if
```

**Comment by nicomazz** [comment]
ops. Fixed!

**Comment by gvnn3** [comment]
Approved but never merged.  @nicomazz @qwattash @arichardson please merge or delete.

**Comment by jrtc27** [comment]
MIPS is gone

**Pull request closed without merging**